traffic_ops/app/lib/API/Profile.pm

```diff
@@ -222,15 +227,25 @@ sub create {
 	if ($existing_desc) {
 		return $self->alert("a profile with the exact same description already exists.");
 	}
-
+	
 	my $cdn = $params->{cdn};
 	my $type = $params->{type};
+	my $routing_disabled = defined($params->{routingDisabled}) ? $params->{routingDisabled} : 0;
+	# Boolean values don't always show properly, so we're going to evaluate these then convert them to standard integers.
+	# This allows the response output to always show true/false correctly.
+	if ($routing_disabled == 1) {
```

Comment:
no @dangogh look at this part. this part makes me want to rethink programming :)

Reply:
I hate it as well, but there's a reason for it. I want the user to be able to use either 1,0,true, or false for the json input, and have it always output true/false in the response.  if the user sends 0 or 1, \$value outputs as true/false but when the user inputs true/false it outputs as a reference.  This basically sets the value to 0 or 1 regardless of the method the user uses in the json.